src/Parallel/Algorithm.hpp

```diff
@@ -105,7 +107,71 @@ template <typename Component>
 using get_charm_base_class_t =
     typename get_charm_base_class<Component,
                                   typename Component::chare_type>::type;
+
+template <typename Metavariables, typename = std::void_t<>>
+struct has_required_phase_sync_function_impl : std::false_type {};
+
+template <typename Metavariables>
+struct has_required_phase_sync_function_impl<
+    Metavariables,
+    std::void_t<decltype(&Metavariables::is_required_sync_phase)>>
+    : std::true_type {};
+
+template <typename Metavariables>
+constexpr bool has_required_phase_sync_function =
+    has_required_phase_sync_function_impl<Metavariables>::value;
+
+template <typename ParallelComponentType, typename ActionType,
+          typename GeneralType>
+struct check_iterable_action_return_type : std::false_type {};
```

Comment:
You are correct that it is not used, but I've added the template parameter so that when the `static_assert` fails, the action that has a bad return type is printed. I'll add a code comment explaining that.